File: backend/routers/portfolio.py
"""Portfolio router — holdings, summary, performance."""
from fastapi import APIRouter

router = APIRouter(prefix="/api/portfolio", tags=["portfolio"])


@router.get("/summary")
def get_summary():
    return {"total_value": 0, "daily_change_pct": 0, "total_gain_loss": 0, "cash": 0}


@router.get("/holdings")
def get_holdings():
    return []


@router.get("/performance")
def get_performance():
    return {"labels": [], "values": []}

File: frontend/src/api/portfolio.ts
import { api } from './client';
import type { Holding, AssetClassBreakdown, PerformancePoint, PortfolioData } from '@fin/shared';

export const portfolioApi = {
  summary: () =>
    api<{
      total_value: number;
      daily_change: number;
      daily_change_pct: number;
      total_return_pct: number;
      cash: number;
    }>('/portfolio/summary'),

  holdings: () => api<Holding[]>('/portfolio/holdings'),

  performance: (period: string = '1Y') =>
    api<PerformancePoint[]>(`/portfolio/performance?period=${period}`),

  assetClasses: () => api<AssetClassBreakdown[]>('/portfolio/asset-classes'),

  full: () => api<PortfolioData>('/portfolio/full'),

  // Phase 39 fix T2.1: probe endpoint — returns { empty: true } when the user has no holdings,
  // so the page can short-circuit to the EmptyState branch synchronously.
  empty: () => api<{ empty: boolean }>('/portfolio/empty'),
};